domain/application/service/unit.go

```diff
@@ -152,29 +157,83 @@ func (s *Service) SetUnitPassword(ctx context.Context, unitName coreunit.Name, p
 // [applicationerrors.ApplicationNotFoundError] if the application doesn't
 // exist. If the unit life is Dead, an error satisfying
 // [applicationerrors.UnitAlreadyExists] is returned.
-func (s *Service) RegisterCAASUnit(ctx context.Context, appName string, args application.RegisterCAASUnitArg) error {
-	if args.PasswordHash == "" {
-		return errors.Errorf("password hash %w", coreerrors.NotValid)
+func (s *ProviderService) RegisterCAASUnit(
+	ctx context.Context, params application.RegisterCAASUnitParams,
+) (coreunit.Name, string, error) {
+	if params.ProviderID == "" {
+		return "", "", errors.Errorf("provider id %w", coreerrors.NotValid)
 	}
-	if args.ProviderID == "" {
-		return errors.Errorf("provider id %w", coreerrors.NotValid)
+
+	pass, err := password.RandomPassword()
+	if err != nil {
+		return "", "", errors.Errorf("generating unit password: %w", err)
 	}
-	if !args.OrderedScale {
-		return errors.Errorf("registering CAAS units not supported without ordered unit IDs").Add(coreerrors.NotImplemented)
+	registerArgs := application.RegisterCAASUnitArg{
+		ProviderID:       params.ProviderID,
+		StorageParentDir: application.StorageParentDir,
+		PasswordHash:     s.passwordHash(pass),
 	}
-	if args.UnitName == "" {
-		return errors.Errorf("missing unit name %w", coreerrors.NotValid)
+
+	// We don't support anything other that statefulsets.
+	// It's hard coded here until other deployment types are supported.
+	deploymentType := caas.DeploymentStateful
+	appName := params.ApplicationName
+
+	var unitName coreunit.Name
+	switch deploymentType {
+	case caas.DeploymentStateful:
+		splitPodName := strings.Split(params.ProviderID, "-")
+		ord, err := strconv.Atoi(splitPodName[len(splitPodName)-1])
+		if err != nil {
+			return "", "", errors.Capture(err)
+		}
+		if unitName, err = coreunit.NewNameFromParts(appName, ord); err != nil {
+			return "", "", errors.Capture(err)
+		}
+		registerArgs.UnitName = unitName
+		registerArgs.OrderedId = ord
+		registerArgs.OrderedScale = true
+	default:
+		return "", "", errors.Errorf("deployment type %q %w", deploymentType, coreerrors.NotSupported)
 	}
```

Comment:
Remove this switch, you're just making the code complicated for now, and we can fix this in future use.

Reply:
yeah agreed, thought about doing that when the code was copied from the facade.
it will be a long time before we support anything else